pkg/sources/s3/s3.go

```diff
@@ -216,19 +218,12 @@ func (s *Source) pageChunker(ctx context.Context, client *s3.S3, chunksChan chan
 			return
 		}
 
-		err := sem.Acquire(ctx, 1)
-		if err != nil {
-			s.log.Error(err, "could not acquire semaphore")
-			continue
-		}
-		wg.Add(1)
-		go func(ctx context.Context, wg *sync.WaitGroup, sem *semaphore.Weighted, obj *s3.Object) {
+		s.jobPool.Go(func() error {
 			defer common.RecoverWithExit(ctx)
-			defer sem.Release(1)
-			defer wg.Done()
 
 			if (*obj.Key)[len(*obj.Key)-1:] == "/" {
```

Comment:
**Nit:** if `obj.Key` is a `string` we could do `strings.HasSuffix(*obj.Key, "/")`